src/scales/scale.time.js

```diff
@@ -560,25 +560,26 @@ module.exports = Scale.extend({
 						datasets[i][j] = timestamp;
 					}
 				} else {
-					for (j = 0, jlen = labels.length; j < jlen; ++j) {
-						timestamps.push(labels[j]);
-					}
 					datasets[i] = labels.slice(0);
+					if (!labelsAdded) {
+						timestamps = timestamps.concat(labels);
+						labelsAdded = true;
+					}
 				}
 			} else {
 				datasets[i] = [];
 			}
 		}
 
 		if (labels.length) {
-			// Sort labels **after** data have been converted
-			labels = arrayUnique(labels).sort(sorter);
 			min = Math.min(min, labels[0]);
 			max = Math.max(max, labels[labels.length - 1]);
 		}
 
 		if (timestamps.length) {
-			timestamps = arrayUnique(timestamps).sort(sorter);
+			if (ilen > 1) {
```

Comment:
There might be a valid use case for having disordered labels, for example in scatter chart. 
I updated the first [example pen](https://codepen.io/kurkle/pen/GbmYPa) to demonstrate.

Reply:
I wasn't sure if we'd ever supported that. I've added the sorting back though